content/blog/webrtc-demo.md

```diff
@@ -0,0 +1,192 @@
+
+Now let’s take a look at those screenshots !
+
+![Login&Home](/images/blog/webrtc-demo/login&home.png)
+![In&OnCall](/images/blog/webrtc-demo/in&oncall.png)
+![Video&MultiCalls](/images/blog/webrtc-demo/video&multi.png)
+
+In the last screenshot, you can see the multiple calls display styling. I would like to dig deeper into this feature, and take a look at the code behind it. 
+
+## Zoom on the multiple calls display
+
+
+First, let’s go into the `dialer.js` file. This file contains every function that will build our dialer (from dialing to video calling and receiving calls). On top of this file, we create an empty object : `sessions`, which is a Wazo object you can use to store your call sessions. On the other hand, `currentSession` is initialized and will be used later. 
+
+```js
+  const sessions = {};
+  let currentSession;
+```
+
+Now let’s study the `initializeWebRtc` function. In the Wazo SDK you have a bunch of useful functions to create a Web RTC application. You set up all of the events and write the names of the functions you’ll be using for these interactions (from line 9 to 17).
+
+But the interesting part for the multiple calls handler is in the `Wazo.Phone.ON_REINVITE` event. We set up the `callSession` variable on the corresponding session id (we’re looking for it into our sessions object).
+
+```js
+  const initializeWebRtc = () => {
+    Wazo.Phone.connect({
+      media: {
+        audio: true,
+        video: true,
+      },
+    });
+
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_INCOMING, openIncomingCallModal);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_ACCEPTED, onCallAccepted);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_FAILED, onCallFailed);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_ENDED, onCallEnded);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_HELD, onSessionUpdate);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_RESUMED, onSessionUpdate);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_MUTED, onSessionUpdate);
+    Wazo.Phone.on(Wazo.Phone.ON_CALL_UNMUTED, onSessionUpdate);
+    Wazo.Phone.on(
+      Wazo.Phone.ON_REINVITE,
+      (session, request, updatedCalleeName) => {
+        const callSession = sessions[session.id];
+        if (callSession) {
+          currentSession.realDisplayName = updatedCalleeName;
+
+          onSessionUpdate(currentSession);
+        }
+      },
+    );
+  }
+```
+
+The `updateScenes` function is called every time something is changed in our 
+scene. A scene is a call session : the interlocutor name, the buttons, etc… 
+The commented part is not relevant, so let’s take a look at what is happening from line 9 to 21. We are looking into the `sessions` object and for each session we are fetching : 
+
+* The corresponding session id
+* The session’s associated number
+
+We set up `currentSession` on the `callSession` id if the session is active (we’re on call with this session’s number). 
+
+```js
+  const updateScenes = status => {
+    const noActiveSession = !Object.keys(sessions).length;
+
+    $('#scenes').html('');
```

Comment:
You can explain that we're using jQuery for simplicity, to not overload the demo with a fancy framework.

Reply:
Now I realize I didn't write about the languages used, I will add that !